umbral-pre/src/serde.rs

```diff
@@ -0,0 +1,131 @@
+use core::fmt;
+use core::marker::PhantomData;
+
+use serde::{de, Deserializer, Serializer};
+
+use crate::traits::{DeserializableFromArray, HasTypeName, SerializableToArray};
+
+// We cannot have a generic implementation of Serialize over everything
+// that supports SerializableToArray, so we have to use this helper function
+// and define implementations manually.
+pub(crate) fn serde_serialize<T, S>(obj: &T, serializer: S) -> Result<S::Ok, S::Error>
+where
+    T: SerializableToArray,
+    S: Serializer,
+{
+    if serializer.is_human_readable() {
```

Comment:
Where is this `is_human_readable` defined?  What does it mean?
If the serializer for something "is human readable", we are returning base64?

Reply:
It's defined by the implementers of specific formats (see [docs](https://docs.serde.rs/serde/trait.Serializer.html#method.is_human_readable)), e.g. JSON, BSON etc. They decide whether it's supposed to be "human-readable" or not. 

> If the serializer for something "is human readable", we are returning base64?

Yep. Although I want to make it a hex representation in case of public keys, the same as it currently works in `nucypher`.